core_v14/src/substrate_value/substrate_value.rs

```diff
@@ -0,0 +1,375 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+// This file is part of substrate-desub.
+//
+// substrate-desub is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// substrate-desub is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with substrate-desub.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Stucture for registering substrate types
+//! Generic SubstrateType enum
+//! Serialization and Deserialization Implementations (to serialize as if it were a native type)
+//! Display Implementation
+
+use super::remote;
+use primitives::crypto::AccountId32;
+use primitives::crypto::{Ss58AddressFormat, Ss58Codec};
+use serde::{ Serialize, Deserialize };
+use std::{convert::TryFrom, fmt};
+
+pub use super::data::Data;
+
+pub type Address = runtime_primitives::MultiAddress<AccountId32, u32>;
+pub type Vote = pallet_democracy::Vote;
+pub type Conviction = pallet_democracy::Conviction;
+
+/// A substrate value which can be instantiated at runtime (normally
+/// through being decoded from SCALE encoded bytes).
+#[derive(Debug, PartialEq, Clone, Serialize)]
+#[serde(untagged)]
+pub enum SubstrateValue {
```

Comment:
How tightly coupled is this type to v14? Could it be shared with the legacy versions?